exporter/exporterhelper/resource_to_label.go

```diff
@@ -32,68 +31,52 @@ func defaultResourceToTelemetrySettings() ResourceToTelemetrySettings {
 	}
 }
 
-// convertResourceToLabels converts all resource attributes to metric labels
-func convertResourceToLabels(md pdata.Metrics) pdata.Metrics {
+// convertResourceToAttributes converts all resource attributes to metric labels
+func convertResourceToAttributes(md pdata.Metrics) pdata.Metrics {
 	cloneMd := md.Clone()
 	rms := cloneMd.ResourceMetrics()
 	for i := 0; i < rms.Len(); i++ {
 		resource := rms.At(i).Resource()
 
-		labelMap := extractLabelsFromResource(&resource)
-
 		ilms := rms.At(i).InstrumentationLibraryMetrics()
 		for j := 0; j < ilms.Len(); j++ {
 			ilm := ilms.At(j)
 			metricSlice := ilm.Metrics()
 			for k := 0; k < metricSlice.Len(); k++ {
 				metric := metricSlice.At(k)
-				addLabelsToMetric(&metric, labelMap)
+				addAttributesToMetric(&metric, resource.Attributes())
 			}
 		}
 	}
 	return cloneMd
 }
 
-// extractAttributesFromResource extracts the attributes from a given resource and
-// returns them as a StringMap.
-func extractLabelsFromResource(resource *pdata.Resource) pdata.StringMap {
-	labelMap := pdata.NewStringMap()
-
-	attrMap := resource.Attributes()
-	attrMap.Range(func(k string, av pdata.AttributeValue) bool {
-		stringLabel := tracetranslator.AttributeValueToString(av)
-		labelMap.Upsert(k, stringLabel)
-		return true
-	})
-	return labelMap
-}
-
-// addLabelsToMetric adds additional labels to the given metric
-func addLabelsToMetric(metric *pdata.Metric, labelMap pdata.StringMap) {
+// addAttributesToMetric adds additional labels to the given metric
+func addAttributesToMetric(metric *pdata.Metric, labelMap pdata.AttributeMap) {
 	switch metric.DataType() {
 	case pdata.MetricDataTypeGauge:
-		addLabelsToNumberDataPoints(metric.Gauge().DataPoints(), labelMap)
+		addAttributesToNumberDataPoints(metric.Gauge().DataPoints(), labelMap)
 	case pdata.MetricDataTypeSum:
-		addLabelsToNumberDataPoints(metric.Sum().DataPoints(), labelMap)
+		addAttributesToNumberDataPoints(metric.Sum().DataPoints(), labelMap)
 	case pdata.MetricDataTypeHistogram:
```

Comment:
Please create a bug that Summary is not supported.

Reply:
Added issue here: https://github.com/open-telemetry/opentelemetry-collector/issues/3817